server/src/internalClusterTest/java/org/elasticsearch/search/functionscore/QueryRescorerIT.java

```diff
@@ -149,33 +150,24 @@ public void testRescorePhrase() throws Exception {
                     5
                 ),
             response -> {
-                assertThat(response.getHits().getTotalHits().value(), equalTo(3L));
+                assertHitCount(response, 3);
                 assertThat(response.getHits().getMaxScore(), equalTo(response.getHits().getHits()[0].getScore()));
-                assertThat(response.getHits().getHits()[0].getId(), equalTo("1"));
-                assertThat(response.getHits().getHits()[1].getId(), equalTo("3"));
-                assertThat(response.getHits().getHits()[2].getId(), equalTo("2"));
+                assertFirstHit(response, hasId("1"));
+                assertSecondHit(response, hasId("3"));
+                assertThirdHit(response, hasId("2"));
             }
         );
-        assertResponse(
+        assertResponses(response -> {
+            assertHitCount(response, 3);
+            assertThat(response.getHits().getMaxScore(), equalTo(response.getHits().getHits()[0].getScore()));
```

Comment:
I would assume that this is safe, but this is a new addition for this test block, right?

Reply:
Just noticed it below; in L#174 :)